Misc improvements to CD pipeline

poltak opened the pull request:
## Misc. fixes:

- [x]  ~not being able to push version bump commits~
    - [x]  ~iOS: `git push ssh-origin master` returns "Everything up-to-date"~
        - does a commit really get made?
    - [x]  ~Android: `/home/travis/.ssh/config line 9: Missing argument.` when pushing~
        - same config as iOS VM, though different OSes
        - are they both using the same OpenSSH version? Default options the same?
    - How important is this? we'll still need some way to manually update version numbers. Also need to deal with not re-triggering CD when these commits are pushed. **Perhaps a better approach is simplifying the steps needed to follow to trigger a CD build**
    - **Conclusion:** simplified the trigger build instructions to a single file to update version number + code, removed the need for the pipeline to commit
- [ ]  ~Ensure new Android builds get uploaded to Play Store production track rather than beta track~
  - _comment from notion to @blackforestboi_ : This might depend on how you want to do releases. From what I understand, it currently works like: CD pipeline uploads new builds to the beta track, which are released straight away, and won't go out to production unless you manually log in and press the button. 
Updating this would instead make the CD pipeline upload new builds straight to the production track, and I think release them straight away
- [x]  still run test CI stages on normal branches

## Misc. improvements:

- [x]  ~Update instructions:~

[CD Pipeline instructions](https://www.notion.so/CD-Pipeline-instructions-2050caf8823948f6beb296129224f27a)

- [x]  ~get a way to be able to test the pipeline locally~
    - development is extremely painful having to actually push new commits to our repo and wait for CI's slower VMs to run through things
    - solution detailed here, essentially just trying to find out which setup is being used on our Travis builds and running that container locally, mimicking commands: [https://stackoverflow.com/questions/21053657/how-to-run-travis-ci-locally/49019950#49019950](https://stackoverflow.com/questions/21053657/how-to-run-travis-ci-locally/49019950#49019950)
    - however, we rely on both an Ubuntu and macOS Travis instances to run Android and iOS builds - we can't just spin up a macOS container in docker, so we'd only be able to replicate the Android build env - could just use our own macs
    - it would also require us to write a script to run, mimicking the steps generated from our travis.yml - it won't just read that like the Travis GH service does
    - **Conclusion:** possible to replicate set-up but would require a few days work to all set up nicely. May be better to treat as another task we could schedule for some time in the (hopefully near) future

- [ ]  get build/dep caches working
    - great ideas in [https://medium.com/@tjkangs/travisci-circleci-2-0-with-fastlane-for-react-native-both-ios-and-android-3f99b71b8691](https://medium.com/@tjkangs/travisci-circleci-2-0-with-fastlane-for-react-native-both-ios-and-android-3f99b71b8691)
    - [ ]  iOS:
        - [ ]  pods
        - [ ]  yarn
        - [ ]  build
    - [x]  android
        - [ ]  gem - could not get working; ended up encountering issues, like running out of OS file watchers during build
        - [x]  yarn
        - [x]  build
- [ ]  get separate beta builds working on develop branch (see other task: https://www.notion.so/worldbrain/Add-separate-app-for-develop-branch-with-staging-environment-keys-so-we-can-both-have-a-production-4aebcdf84eae4a31960f61bac39ca0c6)

blackforestboi commented:
>  Ensure new Android builds get uploaded to Play Store production track rather than beta track

I think having the step of needing to verify via the play and app store, especially for production build is super important. We had a couple of issues that only appeared in the productino track, so until further notice I'd like to have the ability to check on the test of each build before it goes out.

poltak commented:
Alright going to merge this one in now. Tests and linting should be working now for non-master branches (including this). Have made some improvements to caching, mainly with the Android stage, so there should be some overall improvement to deploy speed (depending on if new deps have been daded) but leaving the others now to move onto more important tasks.

@blackforestboi the main thing to note from these changes are that the CD deployment instructions have been updated. Simplified the steps to only require updating 2 values in `package.json`, and added an optional (but recommended) 2nd step detailing how to create a git tag for each release: 
https://www.notion.so/worldbrain/CD-Pipeline-instructions-2050caf8823948f6beb296129224f27a
